src/test/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistencyTest.java

```diff
@@ -105,4 +119,33 @@ void shouldCloseReplicaConnectionsWhenNotConsistent() throws SQLException {
             verify(replica).close();
         }
     }
+
+    @Test
+    void shouldNotOpenConnectionIfNotNeeded() {
+        //given
+        mockReplicaConnections(5, true);
+        final Supplier<Connection> connectionSupplier = mock(Supplier.class);
+        sut = AuroraMultiReplicaConsistency.builder()
+            .replicaConsistency(new ReplicaConsistency() {
+                @Override
+                public void write(Connection main) {
+
+                }
+
+                @Override
+                public boolean isConsistent(Supplier<Connection> replica) {
+                    // Don't use the supplier
```

Comment:
what it it will use supplier?
should it also close it within `isConsistent` ?

Reply:
ReplicaConsistency uses `Supplier`, but it's most likely a connection pool under the hood. We can fix this semantic issue in the next major release by changing `Supplier` into `ConnectionPool`. The connection pool would define who closes the connection.

Reply:
👍

Reply:
@wyrzyk the suggestion to change to `ConnectionPool` never made it out right? Do you know if this was tracked anywhere else besides this comment?

Reply:
No. I started working on the API changes but never finished because we had to hand over the project. (https://github.com/atlassian-labs/db-replica/pull/147/files).